tracer/src/Datadog.Trace/TracerManagerFactory.cs

```diff
@@ -349,13 +352,80 @@ protected virtual ISpanSampler GetSpanSampler(TracerSettings settings)
+            {
+                try
+                {
+                    var telemetrySettings = TelemetrySettings.FromSource(GlobalConfigurationSource.Instance, TelemetryFactory.Config, settings, isAgentAvailable: null);
+                    TelemetryClientConfiguration? telemetryClientConfiguration = null;
+
+                    // We don't know how to handle telemetry in Agentless mode yet
+                    // so we disable telemetry in this case
+                    if (telemetrySettings.TelemetryEnabled && telemetrySettings.Agentless == null)
+                    {
+                        telemetryClientConfiguration = new TelemetryClientConfiguration
+                        {
+                            Interval = (ulong)telemetrySettings.HeartbeatInterval.Milliseconds,
+                            RuntimeId = new CharSlice(Tracer.RuntimeId),
+                            DebugEnabled = telemetrySettings.DebugEnabled
+                        };
+                    }
+
+                    // When APM is disabled, we don't want to compute stats at all
+                    // A common use case is in Application Security Monitoring (ASM) scenarios:
+                    // when APM is disabled but ASM is enabled.
+                    var clientComputedStats = !settings.StatsComputationEnabled && !settings.ApmTracingEnabled;
+
+                    using var configuration = new TraceExporterConfiguration
+                    {
+                        Url = GetUrl(settings),
+                        TraceVersion = TracerConstants.AssemblyVersion,
+                        Env = settings.Environment,
+                        Version = settings.ServiceVersion,
+                        Service = settings.ServiceName,
+                        Hostname = HostMetadata.Instance.Hostname,
+                        Language = ".NET",
+                        LanguageVersion = FrameworkDescription.Instance.ProductVersion,
+                        LanguageInterpreter = FrameworkDescription.Instance.Name,
+                        ComputeStats = settings.StatsComputationEnabled,
+                        TelemetryClientConfiguration = telemetryClientConfiguration,
+                        ClientComputedStats = clientComputedStats
+                    };
+
+                    return new TraceExporter(configuration);
+                }
+                catch (Exception ex)
+                {
+                    Log.Error(ex, "Failed to create native Trace Exporter, falling back to managed API");
+                }
+            }
+
+            return new Api(apiRequestFactory, statsd, updateSampleRates, partialFlushEnabled);
+        }
+
+        private string GetUrl(TracerSettings settings)
+        {
+            switch (settings.Exporter.TracesTransport)
+            {
+                case TracesTransportType.WindowsNamedPipe:
+                    return $"windows://./pipe/{settings.Exporter.TracesPipeName}";
```

Comment:
interesting... the `windows` prefix is _not_ one that we support in the tracer. Is it supported by the agent etc? If so, we probably _should_ support it in theory 🤔